azure/managed-certificate/main.tf

```diff
@@ -0,0 +1,67 @@
+resource "azurerm_key_vault" "main" {
+  name                        = var.name
+  location                    = data.azurerm_resource_group.main.location
+  resource_group_name         = data.azurerm_resource_group.main.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_retention_days  = 7
+  purge_protection_enabled    = false
+  sku_name                    = "standard"
+}
+
+# Give KV secret permissions to the service principal that runs the Terraform apply itself
+resource "azurerm_key_vault_access_policy" "terraform" {
+  key_vault_id = azurerm_key_vault.main.id
+
+  tenant_id = data.azurerm_client_config.current.tenant_id
+  object_id = data.azurerm_client_config.current.object_id
+
+  certificate_permissions = [
+    "Create",
+    "Delete",
+    "DeleteIssuers",
+    "Get",
+    "GetIssuers",
+    "Import",
+    "List",
+    "ListIssuers",
+    "ManageContacts",
+    "ManageIssuers",
+    "Purge",
+    "SetIssuers",
+    "Update"
+  ]
+
+  secret_permissions = [
+    "Backup",
+    "Delete",
+    "Get",
+    "List",
+    "Purge",
+    "Restore",
+    "Restore",
+    "Set"
+  ]
+}
+
+# Give KV secret read permissions to the service
+# Most likely, this is the gateway service
+resource "azurerm_key_vault_access_policy" "service" {
+  key_vault_id = azurerm_key_vault.main.id
+
+  tenant_id = data.azurerm_client_config.current.tenant_id
+  object_id = var.gateway_identity_principal_id
+
+  # the certificate is accessed like a secret...
+  secret_permissions = [
+    "Get", "List"
+  ]
+}
+
+# Often, right after the key vault is created, it's not ready.
+# Waiting for a certain period of time for it to be ready, is advice from issues on GH.
+resource "time_sleep" "wait_240_seconds" {
```

Comment:
Is one of the resources waiting on KV to provision? Or did the GH issue state that it's best to wait a couple of minutes until after KV is created to be deemed "good".

Thought process is if another resource is depending on KV to be ready, do we want to add a `depends_on` block to that resource linked to this?

Reply:
V good question.

I tried a depends_on.. at least I think I did 1 time. Regardless, you're 💯 right that this is gross and we can do better. I'd love to spend an afternoon just messing with this one resource, and optimizing that. One day. :)